internal/sshserver/serverImpl.go

```diff
@@ -86,12 +86,13 @@ func (s *serverImpl) RunWithLifecycle(lifecycle service.Lifecycle) error {
 		s.wg.Add(1)
 		go s.handleConnection(tcpConn)
 	}
+
 	lifecycle.Stopping()
 	s.shuttingDown = true
 	allClientsExited := make(chan struct{})
 	shutdownHandlerExited := make(chan struct{}, 1)
+	s.disconnectClients(lifecycle, allClientsExited)
 	go s.shutdownHandlers.Shutdown(lifecycle.ShutdownContext())
-	go s.disconnectClients(lifecycle, allClientsExited)
```

Comment:
If this does not work, this is actually a bug. The shutdown handlers should get a chance to run before the clients are disconnected forcefully.

Reply:
This seems to work the way it's intended currently: `disconnectClients()` waits until the context timeout (hardcoded 30 seconds currently in main) and then force-disconnects all clients. All this PR does is make this delay configurable.

I delay the shutdown handler until all clients are disconnected because running them means shutting down things like the metrics server which may be needed if the delay is too long, we want to know what happened during that time.

Reply:
In the future it may be beneficial to add a message when a shutdown is triggered prompting all users to re-connect.